src/qutip_qip/decompose/_utility.py

```diff
@@ -29,3 +30,180 @@ def check_gate(gate, num_qubits):
         raise ValueError("Input is not unitary.")
     if gate.dims != [[2] * num_qubits] * 2:
         raise ValueError(f"Input is not a unitary on {num_qubits} qubits.")
+
+
+def _binary_sequence(num_qubits):
+    """ Defines the binary sequence list for basis vectors of a n-qubit gate.
+
+    The string at index `i` is also the row/column index for a basis vector in
+    a numpy array.
+    """
+    old_sequence = ['0', '1']
+    full_binary_sequence = []
+
+    if num_qubits == 1:
+        full_binary_sequence = old_sequence
+    else:
+        for x in range(num_qubits-1):
+            full_binary_sequence = []
+            zero_append_sequence = ['0' + x for x in old_sequence]
+            full_binary_sequence.extend(zero_append_sequence)
+            one_append_sequence = ['1' + x for x in old_sequence]
+            full_binary_sequence.extend(one_append_sequence)
+            old_sequence = full_binary_sequence
+
+    return(full_binary_sequence)
+
+
+def _gray_code_sequence(num_qubits, output_form=None):
```

Comment:
I feel like there is a more efficient way to generate the sequence without first compute a list of all binaries and then copy. But the overhead compared to the overall decomposition is not that big. So let's keep that in mind with a comparably low priority.

Reply:
@BoxiLi Would it be better if we push this later down the line ? I can create an issue for this to remember to work on it. This is because I will need to look over a couple of references to find the most efficient method. 

If it's preferred, I could also add some of the referenced methods below and provide a choice for how the gray codes are generated. 

- References on page 8-9 of https://arxiv.org/pdf/1706.00852.pdf
- References on page 6 of https://arxiv.org/pdf/1608.06001.pdf
- Overview of gray codes - [Ref 1](https://tel.archives-ouvertes.fr/tel-01203339/document), [Ref 2](https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.109.6592&rep=rep1&type=pdf)